cdap-security/src/test/java/co/cask/cdap/security/securestore/FileSecureStoreTest.java

```diff
@@ -0,0 +1,148 @@
+  }
+
+  @Test
+  public void testList_empty() throws IOException {
+    Assert.assertEquals(new HashMap<String, String>(), secureStore.list());
+  }
+
+  @Test
+  public void testList() throws IOException {
+    populateStore();
+    Map<String, String> expectedList = new HashMap<>(2);
+    expectedList.put(KEY1, DESCRIPTION1);
+    expectedList.put(KEY2, DESCRIPTION2);
+    Assert.assertEquals(expectedList, secureStore.list());
+  }
+
+  @Test
+  public void testGet() throws IOException {
+    populateStore();
+    SecureStoreMetadata metadata = FileSecureStoreMetadata.of(KEY1, PROPERTIES_1);
+    SecureStoreData secureStoreData = new FileSecureStoreData(metadata, VALUE1.getBytes(Charsets.UTF_8));
+    Assert.assertArrayEquals(secureStoreData.get(), secureStore.get(KEY1).get());
+    Assert.assertEquals(metadata.getDescription(), secureStore.get(KEY1).getMetadata().getDescription());
+    Assert.assertEquals(metadata.getName(), secureStore.get(KEY1).getMetadata().getName());
+  }
+
+  @Test
+  public void testGetMetadata() throws IOException {
+    populateStore();
+    SecureStoreMetadata metadata = FileSecureStoreMetadata.of(KEY1, PROPERTIES_1);
+    Assert.assertEquals(metadata.getDescription(), secureStore.get(KEY1).getMetadata().getDescription());
+    Assert.assertEquals(metadata.getName(), secureStore.get(KEY1).getMetadata().getName());
+    SecureStoreMetadata metadata2 = FileSecureStoreMetadata.of(KEY2, PROPERTIES_2);
+    Assert.assertEquals(metadata2.getDescription(), secureStore.get(KEY2).getMetadata().getDescription());
+    Assert.assertEquals(metadata2.getName(), secureStore.get(KEY2).getMetadata().getName());
+  }
+
+  @Test
+  public void testOverwrite() throws IOException {
+    secureStoreManager.put(KEY1, VALUE1.getBytes(Charsets.UTF_8), PROPERTIES_1);
+    SecureStoreData oldData = secureStore.get(KEY1);
+    long oldCreateTime = oldData.getMetadata().getLastModifiedTime();
+    Assert.assertArrayEquals(VALUE1.getBytes(Charsets.UTF_8), oldData.get());
+    String newVal = "New value";
+    secureStoreManager.put(KEY1, newVal.getBytes(Charsets.UTF_8), PROPERTIES_1);
+    long newCreateTime = secureStore.get(KEY1).getMetadata().getLastModifiedTime();
+    Assert.assertArrayEquals(newVal.getBytes(Charsets.UTF_8), secureStore.get(KEY1).get());
+    Assert.assertNotEquals(oldCreateTime, newCreateTime);
+  }
+
+  @Test(expected = IOException.class)
+  public void testGet_NonExistent() throws IOException {
+    secureStore.get("Dummy");
+  }
+
+  @Test
+  public void testDelete() throws IOException {
+    populateStore();
+    SecureStoreMetadata metadata = FileSecureStoreMetadata.of(KEY1, PROPERTIES_1);
+    SecureStoreData secureStoreData = new FileSecureStoreData(metadata, VALUE1.getBytes(Charsets.UTF_8));
+    Assert.assertArrayEquals(secureStoreData.get(), secureStore.get(KEY1).get());
+    secureStoreManager.delete(KEY1);
+    try {
+      secureStore.get(KEY1);
```

Comment:
put an `Assert.fail(...)` after the call to get.
Otherwise, the try block may not even throw an exception, and you wouldn't know, but this test case expects there to be an exception thrown.